packages/commerce-sdk-react/src/hooks/ShopperProducts/query.ts

```diff
@@ -31,11 +68,46 @@ export const useProducts = (
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperproducts.shopperproducts-1.html#getproduct} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-export const useProduct = (
-    arg: Argument<Client['getProduct']>
-): UseQueryResult<DataType<Client['getProduct']>, Error> => {
+type UseProductParameters = NonNullable<Argument<Client['getProduct']>>['parameters']
+type UseProductHeaders = NonNullable<Argument<Client['getProduct']>>['headers']
+function useProduct(
+    parameters: UseProductParameters,
+    options?: {
+        headers: UseProductHeaders
+        rawResponse: false
+        queryOptions: UseQueryOptions<DataType<Client['getProduct']> | Response, Error>
+    }
+): UseQueryResult<DataType<Client['getProduct']>, Error>
+function useProduct(
+    parameters: UseProductParameters,
+    options?: {
+        headers: UseProductHeaders
+        rawResponse: true
+        queryOptions: UseQueryOptions<DataType<Client['getProduct']> | Response, Error>
+    }
+): UseQueryResult<Response, Error>
+function useProduct(
+    parameters: UseProductParameters,
+    options?: {
+        headers: UseProductHeaders
+        rawResponse: boolean
+        queryOptions: UseQueryOptions<DataType<Client['getProduct']> | Response, Error>
+    }
+): UseQueryResult<DataType<Client['getProduct']> | Response, Error>
+function useProduct(
+    parameters: UseProductParameters,
+    options?: {
+        headers: UseProductHeaders
+        rawResponse: boolean
+        queryOptions: UseQueryOptions<DataType<Client['getProduct']> | Response, Error>
+    }
+): UseQueryResult<DataType<Client['getProduct']> | Response, Error> {
```

Comment:
Can we safely delete the duplicate signature?

```suggestion
function useProduct(
    parameters: UseProductParameters,
    options?: {
        headers: UseProductHeaders
        rawResponse: boolean
        queryOptions: UseQueryOptions<DataType<Client['getProduct']> | Response, Error>
    }
): UseQueryResult<DataType<Client['getProduct']> | Response, Error> {
```

Reply:
There are other hooks with similar duplicate signatures. Can you check them too please?